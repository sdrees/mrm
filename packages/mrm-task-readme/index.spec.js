jest.mock('fs');
jest.mock('mrm-core/src/util/log', () => ({
	added: jest.fn(),
}));

const fs = require.requireActual('fs');
const path = require('path');
const { omitBy } = require('lodash');
const { getConfigGetter } = require('mrm');
const vol = require('memfs').vol;
const task = require('./index');

const stringify = o => JSON.stringify(o, null, '  ');

afterEach(() => vol.reset());

it('should add a readme', () => {
	vol.fromJSON({
		[`${__dirname}/templates/Readme.md`]: fs
			.readFileSync(path.join(__dirname, 'templates/Readme.md'))
			.toString(),
		'/package.json': stringify({
			name: 'unicorn',
		}),
	});

	task(
		getConfigGetter({
			name: 'Gendalf',
			url: 'https://middleearth.com',
			github: 'gendalf',
		})
	);

	expect(
		omitBy(vol.toJSON(), (v, k) => k.startsWith(__dirname))
	).toMatchSnapshot();
});
