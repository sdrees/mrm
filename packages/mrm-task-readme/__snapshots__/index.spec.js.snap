// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should add a readme 1`] = `
Object {
  "/Readme.md": "# unicorn

...

## Change log

The change log can be found on the [Releases page](https://github.com/gendalf/unicorn/releases).

## Contributing

Everyone is welcome to contribute. Please take a moment to review the [contributing guidelines](Contributing.md).

## Authors and license

[Gendalf](http://middleearth.com) and [contributors](https://github.com/gendalf/unicorn/graphs/contributors).

MIT License, see the included [License.md](License.md) file.
",
  "/package.json": "{
  \\"name\\": \\"unicorn\\"
}",
}
`;
